### ISSUE TITLE
Reorganize the list of "Apps"

### ISSUE THREAD

**Issue by EVOTk** [gotify/server#358]
**Is your feature request related to a problem? Please describe.**
On the side, the applications are displayed in the order of creation. I wish I could change this order.

**Describe the solution you'd like**
In Apps, choose an order number for each "Apps"

**Additional context**
![2020-09-10 21_51_26-Gotify](https://user-images.githubusercontent.com/45015615/92794357-d36a3200-f3af-11ea-9869-8f565995baf7.png)
For example, I would like to be able to put "EVOserv" at the top of the list, but for that it would have been necessary to create it first!
Now, for that I will have to delete all the Apps "above", and re-create them, which implies a change of token for this one!

Thx

**Comment by jmattheis** [comment]
Yeah, seems like a good feature. You could also rename the first application, then you don't have to remove all applications.

**Comment by EVOTk** [comment]
Thanks for your feedback :)
Renaming does not solve the problem of the "token" which is linked to an application. Renaming everything is the same as deleting, it will be necessary to change the tokens in the all applications, so that they correspond again.

**Comment by narol69** [comment]
I like the suggestion of EVOTk.

**Comment by junpet** [comment]
Can I hop on the train to ask optional sorting by alphabet?

**Comment by nocturnalmc** [comment]
+1 for this. Currently using gotify to manage 7 apps. Would like to rearrange my application list

**Comment by Raytestrak** [comment]
Bump to an old thread, it can be done via the database. Just reorder the IDs of the applications.

**Comment by xavier-GitHub76** [comment]
The sort is OK with : 
const apps = appStore.getItems().slice().sort((a,b) => a.name.localeCompare(b.name));

Tests OK : 

Applications created in this order :
- Apple
- Orange
- Banana

Items are sorted on screen : 
![image](https://github.com/gotify/server/assets/82046143/2d61123b-404b-45ae-b09c-e1d9dd1e2f36)

**Comment by samcro1967** [comment]
> Bump to an old thread, it can be done via the database. Just reorder the IDs of the applications.

Here is a script to do this automatically:

**gotify_sort_apps_by_names.sh**
```
#!/bin/bash

# Define the database file path
DB_PATH="gotify.db"

# Step 1: Rename the current applications table to applications_old
sqlite3 "$DB_PATH" "ALTER TABLE applications RENAME TO applications_old;"

# Step 2: Create a new applications table with the same schema
sqlite3 "$DB_PATH" "CREATE TABLE applications (id INTEGER PRIMARY KEY AUTOINCREMENT, token VARCHAR(180), user_id INTEGE>

# Step 3: Insert sorted data from applications_old into the new applications table with updated IDs
sqlite3 "$DB_PATH" "INSERT INTO applications (id, token, user_id, name, description, internal, image, default_priority,>

# Step 4: Drop the applications_old table
sqlite3 "$DB_PATH" "DROP TABLE applications_old;"

echo "Applications table IDs updated successfully."
```

**Comment by NightHawkATL** [comment]
> #!/bin/bash
> 
> # Define the database file path
> DB_PATH="gotify.db"
> 
> # Step 1: Rename the current applications table to applications_old
> sqlite3 "$DB_PATH" "ALTER TABLE applications RENAME TO applications_old;"
> 
> # Step 2: Create a new applications table with the same schema
> sqlite3 "$DB_PATH" "CREATE TABLE applications (id INTEGER PRIMARY KEY AUTOINCREMENT, token VARCHAR(180), user_id INTEGE>
> 
> # Step 3: Insert sorted data from applications_old into the new applications table with updated IDs
> sqlite3 "$DB_PATH" "INSERT INTO applications (id, token, user_id, name, description, internal, image, default_priority,>
> 
> # Step 4: Drop the applications_old table
> sqlite3 "$DB_PATH" "DROP TABLE applications_old;"
> 
> echo "Applications table IDs updated successfully."

I just tried this while exec'd into the container and it doesn't like the sqlite3 command for some reason.
![image](https://github.com/user-attachments/assets/bb8b0f24-fb0d-4f28-baa5-88b3af5f29c4)

**Comment by samcro1967** [comment]
I did this from my host, not from inside the container.  Need to make sure you have sqlite3 installed on your host.

**Comment by NightHawkATL** [comment]
I don't have that.
![image](https://github.com/user-attachments/assets/c45f0630-c95d-4d7d-9f77-112e64108c84)

**Comment by samcro1967** [comment]
Let's try this:

[gotify_sort_apps_by_names.txt](https://github.com/user-attachments/files/16789390/gotify_sort_apps_by_names.txt)

Download the file above and rename the file extension from ".txt" to ".sh".  Make executable and then run it.

**Comment by NightHawkATL** [comment]
finally worked. thank you! My OCD is happy now.

**Comment by samcro1967** [comment]
You're welcome.  Glad it finally worked for you.  Us OCD'ers have to stick together...

**Comment by NightHawkATL** [comment]
I loaded the script onto my github and linked to you for the credit. I gave step-by-step instructions on how to use it. Linked to this thread as well for the discussion. Hope you don't mind. Gave you all the credit.

**Comment by ldidry** [comment]
There is a problem with the script: it only works with sqlite. PostgreSQL, for ex. does not output records sorted by id unless told to do it.

There should be some code either server-side or client-side to make sure you always have the same order (for a first step, sorting differently could be done in a second time)

**Comment by jmattheis** [comment]
Yes.

**Comment by jberlyn** [comment]
I have just created a pull request that aims to solve this issue. It adds a Sort Order column to Applications so that you can define your own sort order. The UI will order the list of Applications by the sort order ascending.

https://github.com/gotify/server/pull/804

**Comment by OdinVex** [comment]
> I have just created a pull request that aims to solve this issue. It adds a Sort Order column to Applications so that you can define your own sort order. The UI will order the list of Applications by the sort order ascending.
> 
> [#804](https://github.com/gotify/server/pull/804)

Awesome, hope to see it. :)

**Comment by xavier-GitHub76** [comment]
I will push a pull request today to add drag and drop on jberlyn's sortOrder

**Comment by xavier-GitHub76** [comment]
PR is created (https://github.com/gotify/server/pull/856)
